src/test/java/org/cyclonedx/maven/BaseCycloneDxMojoTest.java

```diff
@@ -0,0 +1,132 @@
+package org.cyclonedx.maven;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.maven.model.Developer;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.cyclonedx.model.Component;
+import org.cyclonedx.model.Dependency;
+import org.cyclonedx.model.OrganizationalContact;
+import org.cyclonedx.model.OrganizationalEntity;
+import org.cyclonedx.model.organization.PostalAddress;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class BaseCycloneDxMojoTest {
+
+    /**
+     * Minimal test class that can create an instance of BaseCycloneDxMojo
+     * so we can do some unit testing.
+     */
+    private static class BaseCycloneDxMojoImpl extends BaseCycloneDxMojo {
+
+        @Override
+        protected String extractComponentsAndDependencies(Set<String> topLevelComponents,
+            Map<String, Component> components, Map<String, Dependency> dependencies) throws MojoExecutionException {
+            return "";
+        }
+    }
+
+    @Test
+    @DisplayName("Verify that the default configuration does not have manufacturer information ")
+    void createManufacturer() {
+        BaseCycloneDxMojoImpl mojo = new BaseCycloneDxMojoImpl();
+        OrganizationalEntity manufacturer = mojo.createManufacturer(null, null);
+        assertNotNull(manufacturer);
```

Comment:
Should this be `null` if there is no data?

Reply:
Security check if you have a very minimal POM file without anything. Could be removed, since checks are in place before calling the tested function, but I didn't see any unit testing at all to the `BaseCycloneDxMojo`, only some testing with POM files running with the super classes. Writing unit test for `execute()` function will take lot of time.

```java
                if (organization != null || (developers != null && !developers.isEmpty())) {
                    metadata.setManufacturer(createManufacturer(organization, developers));
                }
```